modules/workflow/src/main/java/org/eclipse/xpanse/modules/workflow/handle/migrate/MigrateDeployProcess.java

```diff
@@ -53,18 +55,28 @@ public void execute(DelegateExecution execution) {
         runtimeService.updateBusinessKey(processInstanceId, newId.toString());
 
         int deployRetryNum = (int) variables.get(MigrateConstants.DEPLOY_RETRY_NUM);
-
+        boolean isSendDeploy =
+                Objects.isNull(variables.get(MigrateConstants.IS_SEND_DEPLOY)) ? false :
+                        (boolean) variables.get(MigrateConstants.IS_SEND_DEPLOY);
+        if (!isSendDeploy) {
+            isSendDeploy = deployService.deployService(newId, getDeployRequest(variables));
+            runtimeService.setVariable(processInstanceId, MigrateConstants.IS_SEND_DEPLOY,
+                    isSendDeploy);
+        }
         runtimeService.setVariable(processInstanceId, MigrateConstants.DEPLOY_RETRY_NUM,
                 deployRetryNum + 1);
         if (deployRetryNum > 0) {
             log.info("Process instance: {} retry deployment service : {},RetryNum:{}",
                     processInstanceId, newId, deployRetryNum);
         }
-        MigrateRequest migrateRequest =
-                (MigrateRequest) variables.get(MigrateConstants.MIGRATE_REQUEST);
-        DeployRequest deployRequest = new DeployRequest();
-        BeanUtils.copyProperties(migrateRequest, deployRequest);
-        boolean isDeploySuccess = deployService.deployService(newId, deployRequest);
+        ServiceDeploymentState deploymentState =
+                deployService.getDeployServiceDetails(newId).getServiceDeploymentState();
```

Comment:
if the deployment status is checked immediately, then the deployment is never successful. If i understand it right, we must wait for the webhook update and then proceed with the BPMN next steps. Not sure if this code is really handling that.